### DISCUSSION TITLE
Assign sprint points to a task via API?

### DISCUSSION THREAD

**Discussion by cviebrock** [ComfortablyCoding/clickup.js#35] [Q&A]
I'm writing a script to migrate some stories from Pivotal Tracker over to Clickup, and would like to include the sprint points for each new Task I create in CU.

I've tried adding `points: 5` and/or `sprint_points: 5` to the data I pass to `lists.createTask(id,data)`, but neither of those work.

Is this possible at all?

**Comment by cviebrock** [comment]
I should clarify, I'm using the Sprint Points ClickApp for this.  I have it enabled for the Space in which the List I am trying to create a task in exists. :)

**Comment by ComfortablyCoding** [comment]
Hard for me to say as I can't test it myself. 

The easiest way to see how it should be structured is to view the API response from a task that is already created with a sprint points value. That should give some idea on the expected structure.

Alternatively you can view the networks HTTP requests when you click on adding the sprint point in the UI and see how that is structured.

I had a look at the API and don't see any mention of sprint points so I would assume add-ons are a little different? Hard for me to debug this.

**Comment by cviebrock** [comment]
The response returns a `points: 8` as part of the task structure, but setting that on a createTask request doesn't work -- I just get `null` in the new task.

I viewed the network request as I updated an existing story's points, and see that it makes a request to:

```
PUT https://app.clickup.com/tasks/v1/task/abcd1234/pointsEstimate

{ points: 8 }
```

so I tried updating `src/routes/Tasks.js` by adding:

```js

	/**
	 * Update the sprint points for a task
	 *
	 * @param {String} taskId The task id
	 * @param {Number} points The sprint points estimate
	 */
	async updatePoints(taskId, points) {
		return this._request.put({
			endpoint: `${this.route}/${taskId}/pointsEstimate`,
			params: {points},
		});
	}
```

However, using that function returns the error `Cannot PUT /api/v2/task/2t29utt/pointsEstimate` ... so I'm guessing it's a `v1` API thing only, and not `v2`.  I can't immediately see how to force a request to the `v1` URL.  Is that possible?

**Reply by ComfortablyCoding** [reply]
It is possible this functionality is not exposed for v2, as I mentioned I see no mention of this in the v2 docs.

In terms of setting the URL to use v1 the `Clickup` class accepts a second parameter where the base URL can be set. [By default it is the v2 URL](https://github.com/ComfortablyCoding/clickup.js/blob/f4d12e4c9f4e90d55ec18913dafece3172b8c724/src/utils/createRequestInstance.js#L17).

```js
const { Clickup } = require('clickup.js');
const token = '...'; // API access token
const requestOptions = { prefixUrl:"the v1 base URL" };
const clickup = new Clickup(token, requestOptions);
```
This will set it for all requests, it is possible to do it for just specific requests but that is more advanced.

**Comment by cviebrock** [comment]
I just tried hacking something in by changing the above to:
```js
endpoint: `../../tasks/v1/${this.route}/${taskId}/pointsEstimate`,
```
I'm now hitting the right endpoint (i.e. what matches when I use the Clickup web interface).  But I get an error:
```
{ err: 'Auth header missing', ECODE: 'JWT_008' }
```
So ... I'm guessing v1 of the API uses a different auth method/key than v2?

**Reply by ComfortablyCoding** [reply]
That error code does not look to be [a valid one for the public API](https://jsapi.apiary.io/apis/clickup20/introduction/error-handling.html) this makes me think it is meant for internal uses with user JWTs used for requests.

At this point I would recommend reaching out to Clickup and see what they say, they are usually quite responsive and happy to help. It might just be a current limitation of the v2 API.
